src/cursesport.cpp

```diff
@@ -594,6 +598,19 @@ int clearok(WINDOW *win)
     return 1;
 }
 
+int redrawwin(WINDOW* win)
+{
+    if (! win) {
+        return 0;
+    }
+    win->draw = true;
+    for (int i = 0; i < win->height; i++) {
```

Comment:
> clearok updates  stdscr  without touching  win  and I don't know if it's intended or not, but calling it doesn't fix black screen issues, while looping does.

I did not see that, it looks like a bug. It's only used inside of `wclear`, which in turn is only used by `clear` in cursesport.cpp, it's a mess and it's probably not worth touching it.

The implementation of `redrawwin` is fine.